### PR TITLE
QuickCheck generators for update proposals (protocol parameters and application versions and installer hashes)

### LINKED ISSUE #1034: quickcheck generators for shelley updates
The updates in the shelley exec model are defined here: https://github.com/input-output-hk/cardano-ledger-specs/blob/master/shelley/chain-and-ledger/executable-spec/src/Updates.hs#L87

Write quickcheck generators for this data type.

### PR DESCRIPTION
**Pull request by mgajda**
I just updated my generators of Update to the latest PR.
Please let me know how to improve it.
Should close #1034

### CODE REVIEW

**Comment by erikd** [comment]
Please to not use the default title (which is your branch name) for PRs. The default name means close to nothing and I am unable to figure out if this is something I need to be aware of if you don't give the PR an appropriate title.

**Comment by mgajda** [comment]
@uroboros  _(Force push to rebase on latest master)_

@erikd Title updated. Let me know if it still can be improved.

File: shelley/chain-and-ledger/executable-spec/test/Generator/Update/QuickCheck.hs [hunk 1]
```diff
@@ -1,28 +1,50 @@
...
+                         ,PParamsUpdate(..)
+                         ,PPUpdate(..)
+                         ,SystemTag(..)
+                         ,InstallerHash(..)
+                         ,ApVer(..)
+                         ,ApName(..)
+                         ,Applications(..)
+                         ,Mdt(..)
+                         ,AVUpdate(..)
+                         ,Update(..))
```
Comment (mgudemann) [inline shelley/chain-and-ledger/executable-spec/test/Generator/Update/QuickCheck.hs]: please keep to the formatting used, i.e., multiple lines for imports. I thought that stylish does this automatically ?

Comment (mgajda) [reply shelley/chain-and-ledger/executable-spec/test/Generator/Update/QuickCheck.hs]: @mgudemann I followed the best practice that attempts to minimize diffs in a frequent case when there are many imports, but many of them change.

Do you think we should prefer following `stylish` at all times instead?

Comment (mgudemann) [reply shelley/chain-and-ledger/executable-spec/test/Generator/Update/QuickCheck.hs]: my opinion here is yes, follow automated formatting using stylish, I myself find it easy to skip import changes and redundant or missing imports are caught anyhow using CI

**Pull request merged**